src/utils/__tests__/Cache.test.ts

```diff
@@ -0,0 +1,61 @@
+import Cache from '../Cache';
+
+let cache: Cache;
+
+describe('util/Cache', () => {
+    beforeEach(() => {
+        cache = new Cache();
+    });
+
+    test('should set, get and unset correctly', () => {
+        cache.set<string>('foo', 'bar');
+        expect(cache.has('foo')).toBeTruthy();
+        expect(cache.get<string>('foo')).toBe('bar');
+
+        cache.unset('foo');
+        cache.unset('foo');
+        expect(cache.has('foo')).toBeFalsy();
+    });
+
+    test('should set and get correctly with 0 value', () => {
+        cache.set<number>('foo', 0);
+        expect(cache.has('foo')).toBeTruthy();
+        expect(cache.get<number>('foo')).toBe(0);
+    });
+
+    test('should return undefined when not set', () => {
+        cache.set<string>('foo1', 'bar1');
+        cache.set<string>('foo2', 'bar2');
+        cache.set<string>('foo3', 'bar3');
+        expect(cache.get<string>('foo1')).toBe('bar1');
+        expect(cache.get<string>('foo2')).toBe('bar2');
+        expect(cache.get<string>('foo3')).toBe('bar3');
+        cache.unsetAll('foo');
+        expect(cache.get<string>('foo1')).toBeUndefined();
+        expect(cache.get<string>('foo2')).toBeUndefined();
+        expect(cache.get<string>('foo3')).toBeUndefined();
+    });
+
+    test('should properly unset all', () => {
+        expect(cache.get<string>('foobar')).toBeUndefined();
```

Comment:
Can you set some properties in the cache and call unsetAll before asserting it's behavior?